Detections/MultipleDataSources/SucessfullSiginFromPhingLink.yaml

```diff
@@ -0,0 +1,143 @@
+        )
+    | mv-apply Entity = Entities on
+        (
+        where Entity.Type == 'ip'
+        | extend EntityIp = tostring(Entity.Address)
+        )
+    | mv-apply Entity = Entities on
+        (
+        where Entity.Type == 'host'
+        | extend EntityHost = tostring(Entity.HostName)
+        )
+    | mv-apply Entity = Entities on
+        (
+        where Entity.Type == 'url'
+        | extend EntityUrl = tostring(Entity.Url)
+        )
+    | where not(ipv4_is_private(EntityIp))
+    | summarize
+        AccountUpn=take_any(EntityUPN),
+        IPAddress=take_any(EntityIp),
+        HostName=take_any(EntityHost),
+        Url=tostring(tolower(take_any(EntityUrl))),
+        AlertTime= min(TimeGenerated),
+        make_set(SystemAlertId)
+        by ProductName, AlertName
+    // matching with 3rd party network logs and 3p Alerts 
+    | join kind= inner (CommonSecurityLog
+        | where DeviceVendor has_any  ("Palo Alto Networks", "Fortinet", "Check Point", "Zscaler")
+        | where DeviceAction != "Block"
+        | where isnotempty(DestinationHostName) 
+        | where isnotempty(RequestURL)
+        | project
+            3plogTime=TimeGenerated,
+            DeviceVendor,
+            DeviceProduct,
+            Activity,
+            DestinationHostName,
+            DestinationIP,
+            RequestURL=tostring(tolower(RequestURL)),
+            MaliciousIP,
+            SourceUserName=tostring(tolower(SourceUserName)),
+            IndicatorThreatType,
+            ThreatSeverity,
+            ThreatConfidence,
+            SourceUserID,
+            SourceHostName)
+        on $left.Url == $right.RequestURL
+    // matching successful Login from suspicious IP 
+    | join kind=inner (SigninLogs
+        //filtering the Successful Login
+        | where ResultType == 0
+        | project
+            IPAddress,
+            SourceSystem,
+            SigniningTime= TimeGenerated,
+            OperationName,
+            ResultType,
+            ResultDescription,
+            AlternateSignInName,
+            AppDisplayName,
+            AuthenticationRequirement,
+            ClientAppUsed,
+            RiskState,
+            RiskLevelDuringSignIn,
+            UserPrincipalName=tostring(tolower(UserPrincipalName)),
+            Name = tostring(split(UserPrincipalName, "@")[0]),
+            UPNSuffix =tostring(split(UserPrincipalName, "@")[1]))
+        on $left.DestinationIP == $right.IPAddress and $left.SourceUserName == $right.UserPrincipalName  
+    | where SigniningTime between ((AlertTime - 6h) .. (AlertTime + 6h)) and 3plogTime between ((AlertTime - 6h) .. (AlertTime + 6h))      
+entityMappings:
+    - entityType: Account
+      fieldMappings:
+      - identifier: Name
+        columnName: Name
+      - identifier: UPNSuffix
+        columnName: UPNSuffix
+    - entityType: IP
+      fieldMappings:
+      - identifier: Address
+        columnName: DestinationIP
+    - entityType: DNS
```

Comment:
Is it possible to get the full URL as an entity as well?

Reply:
URL added